api/src/opentrons/hardware_control/modules/mod_abc.py

```diff
@@ -13,15 +22,61 @@ class AbstractModule(abc.ABC):
     @abc.abstractmethod
     async def build(cls,
                     port: str,
-                    interrupt_callback,
-                    simulating: bool = False) -> 'AbstractModule':
+                    interrupt_callback: InterruptCallback,
+                    simulating: bool = False,
+                    loop: asyncio.AbstractEventLoop = None) \
+            -> 'AbstractModule':
         """ Modules should always be created using this factory.
 
         This lets the (perhaps blocking) work of connecting to and initializing
         a module be in a place that can be async.
         """
         pass
 
+    @abc.abstractmethod
+    def __init__(self,
+                 port: str,
+                 simulating: bool = False,
+                 loop: asyncio.AbstractEventLoop = None) -> None:
+        self._port = port
+        if None is loop:
+            self._loop = asyncio.get_event_loop()
+        else:
+            self._loop = loop
+        self._device_info = None
+        self._available_update_version: Optional[str] = None
+        self._available_update_path = self.get_bundled_fw()
+        return None
+
+    def get_bundled_fw(self) -> Optional[Path]:
+        """ Get absolute path to bundled version of module fw if available. """
+        name_to_fw_file_prefix = {
+            "tempdeck": "temperature-module", "magdeck": "magnetic-module"}
+        name = self.name()
+        file_prefix = name_to_fw_file_prefix.get(name, name)
+        MODULE_FW_RE = re.compile(f'{file_prefix}@v(.*).(hex|bin)')
+        fw_dir = CONFIG['robot_firmware_dir']
+        fw_resources = [fw_dir / item for item in os.listdir(fw_dir)]
```

Comment:
```suggestion
        fw_resources = fw_dir.iterdir()
```

Path objects have this handy operation that returns an iterator, which we then iterate through in line 60 (could even combine the two now)

Reply:
so handy ✋